src/plugins/kibana/server/lib/__tests__/process_es_ingest_simulate_error.js

```diff
@@ -0,0 +1,19 @@
+import processESIngestSimulateError from '../process_es_ingest_simulate_error';
+import expect from 'expect.js';
+import _ from 'lodash';
+
+describe('processESIngestSimulateError', function () {
+
+  it('result will be returned for processor that threw the error', function () {
+    const dirtyProcessorId = 'processor1';
+    const error = _.set({}, 'body.error.reason', 'foobar');
+
+    const expected = [
+      { processorId: 'processor1', error: { compile: true, message: 'foobar' } }
+    ];
+    const actual = processESIngestSimulateError(dirtyProcessorId, error);
+
+    expect(actual).to.eql(expected);
```

Comment:
You probably want to do `expect(_.isEqual(actual, expected)).to.be.ok()` instead.

Expect.js's `eql()` assertion recursively checks loose equality, so weird conditions can pass it like `expect({foo: true}).to.eql({foo: 1})` whereas `expect({foo: true}).to.eql({foo: 2})` will throw an assertion error.

Lodash's isEqual checks for strict equality.